### PR TITLE
prevent libgccjit.so download on unsupported os/arch

### LINKED ISSUE #527: Do not allow downloading libgccjit.so on other systems than Linux x86-64 in the build system

### PR DESCRIPTION
**Pull request by Skgland**
Alternative to #528, I find it easier to show what I meant rather than tell.

Fixing up the download URL while at it. 

fixes #527

### CODE REVIEW

File: build_system/src/config.rs [hunk 1]
```diff
@@ -469,11 +463,28 @@ impl ConfigInfo {
...
     with_progress_bar: bool,
 ) -> Result<(), String> {
+    let url = if std::env::consts::OS == "linux" || std::env::consts::ARCH == "x86_64" {
+        format!("https://github.com/rust-lang/gcc/releases/download/master-{}/libgccjit.so", commit)
+    } else {
+        eprintln!(
+            r"\
+        pre-compiled libgccjit.so not available for this os or architecture\n\
+        please compile it yourself and in the config.toml set\n\
+        `download-gccjit = false` and `gcc-path` to the appropriate directory\
```
Comment (GuillaumeGomez) [inline build_system/src/config.rs:478]:
```suggestion
        Pre-compiled libgccjit.so not available for this os or architecture.\n\
        Please compile it yourself and update the `config.toml` file\n\
        to `download-gccjit = false` and set `gcc-path` to the appropriate directory.\
```

File: build_system/src/config.rs [hunk 2]
```diff
@@ -469,11 +463,28 @@ impl ConfigInfo {
...
+    commit: &str,
     output_dir: &Path,
     tempfile_name: String,
     with_progress_bar: bool,
 ) -> Result<(), String> {
+    let url = if std::env::consts::OS == "linux" || std::env::consts::ARCH == "x86_64" {
+        format!("https://github.com/rust-lang/gcc/releases/download/master-{}/libgccjit.so", commit)
+    } else {
+        eprintln!(
+            r"\
```
Comment (GuillaumeGomez) [inline build_system/src/config.rs:475]: Is the `r` useful here?

File: build_system/src/config.rs [hunk 3]
```diff
@@ -469,11 +463,28 @@ impl ConfigInfo {
...
 
 fn download_gccjit(
-    url: String,
+    commit: &str,
     output_dir: &Path,
     tempfile_name: String,
     with_progress_bar: bool,
 ) -> Result<(), String> {
+    let url = if std::env::consts::OS == "linux" || std::env::consts::ARCH == "x86_64" {
+        format!("https://github.com/rust-lang/gcc/releases/download/master-{}/libgccjit.so", commit)
```
Comment (GuillaumeGomez) [inline build_system/src/config.rs:472]: Thanks! :)

Comment (Skgland) [reply build_system/src/config.rs:472]: Oh, looking at this I just noticed that I messed up the condition it should be and not or.

Comment (GuillaumeGomez) [reply build_system/src/config.rs:472]: Oh indeed! Completely missed it as well. ^^'

File: build_system/src/config.rs (hunk 2 above)
Comment (Skgland) [reply build_system/src/config.rs:475]: Quite the opposite of useful actually as a raw string is not what I want here.🤦‍♂️

**Comment by Skgland** [comment]
Ok, applied your suggestion regarding the message text change, fixed the conditional to be `&&` instead of `||` and changed the raw-string that shouldn't be raw to a normal string.
Assuming CI agrees, this is good to go from my point.

File: build_system/src/config.rs [hunk 4]
```diff
@@ -469,11 +463,28 @@ impl ConfigInfo {
...
 ) -> Result<(), String> {
+    let url = if std::env::consts::OS == "linux" && std::env::consts::ARCH == "x86_64" {
+        format!("https://github.com/rust-lang/gcc/releases/download/master-{}/libgccjit.so", commit)
+    } else {
+        eprintln!(
+            "\
+        Pre-compiled libgccjit.so not available for this os or architecture.\n\
+        Please compile it yourself and update the `config.toml` file\n\
+        to `download-gccjit = false` and set `gcc-path` to the appropriate directory.\
+        "
```
Comment (GuillaumeGomez) [inline build_system/src/config.rs:479]: Just thinking but at this point we could simply do:

```suggestion
Pre-compiled libgccjit.so not available for this os or architecture.
Please compile it yourself and update the `config.toml` file
to `download-gccjit = false` and set `gcc-path` to the appropriate directory."
```

Comment (Skgland) [reply build_system/src/config.rs]: Sure, if that is preferred, done

**Review by GuillaumeGomez — APPROVED** [review]

**Pull request merged**